torchmetrics/functional/image/d_lambda.py

```diff
@@ -0,0 +1,141 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Tuple
+
+import torch
+from torch import Tensor
+from typing_extensions import Literal
+
+from torchmetrics.functional.image.uqi import universal_image_quality_index
+from torchmetrics.utilities.checks import _check_same_shape
+from torchmetrics.utilities.distributed import reduce
+
+
+def _d_lambda_update(ms: Tensor, fused: Tensor) -> Tuple[Tensor, Tensor]:
+    """Updates and returns variables required to compute Spectral Distortion Index. Checks for same shape and type
+    of the input tensors.
+
+    Args:
+        ms: Low resolution multispectral image
+        fused: High resolution fused image
+    """
+
+    if ms.dtype != fused.dtype:
+        raise TypeError(
+            f"Expected `ms` and `fused` to have the same data type. Got ms: {ms.dtype} and fused: {fused.dtype}."
+        )
+    _check_same_shape(ms, fused)
+    if len(ms.shape) != 4:
+        raise ValueError(
+            f"Expected `ms` and `fused` to have BxCxHxW shape. Got ms: {ms.shape} and fused: {fused.shape}."
+        )
+    return ms, fused
+
+
+def _d_lambda_compute(
+    ms: Tensor,
+    fused: Tensor,
+    p: int = 1,
+    reduction: Literal["elementwise_mean", "sum", "none"] = "elementwise_mean",
+) -> Tensor:
+    """Computes Spectral Distortion Index.
+
+    Args:
+        ms: Low resolution multispectral image
+        fused: High resolution fused image
+        p: a parameter to emphasize large spectral difference (default: 1)
+        reduction: a method to reduce metric score over labels.
+
+            - ``'elementwise_mean'``: takes the mean (default)
+            - ``'sum'``: takes the sum
+            - ``'none'``: no reduction will be applied
+
+    Example:
+        >>> import torch
+        >>> _ = torch.manual_seed(42)
+        >>> ms = torch.rand([16, 3, 16, 16])
+        >>> fused = ms * 0.75
+        >>> ms, target = _d_lambda_update(ms, fused)
+        >>> _d_lambda_compute(ms, fused)
+        tensor(3.4769e-08)
+    """
+    if p <= 0:
+        raise ValueError(f"Expected `p` to be a positive integer. Got p: {p}.")
+
+    L = ms.shape[1]
+
+    M1 = torch.zeros((L, L))
+    M2 = torch.zeros((L, L))
+
+    for k in range(L):
+        for r in range(k, L):
+            M1[k, r] = M1[r, k] = universal_image_quality_index(fused[:, k : k + 1, :, :], fused[:, r : r + 1, :, :])
+            M2[k, r] = M2[r, k] = universal_image_quality_index(ms[:, k : k + 1, :, :], ms[:, r : r + 1, :, :])
+
+    diff = torch.pow(torch.abs(M1 - M2), p)
+    # Special case: when number of channels (L) is 1, there will be only one element in M1 and M2. Hence no need to sum.
+    if L == 1:
+        output = torch.pow(diff, (1.0 / p))
+    else:
+        output = torch.pow(1.0 / (L * (L - 1)) * torch.sum(diff), (1.0 / p))
+    return reduce(output, reduction)
+
+
+def spectral_distortion_index(
+    ms: Tensor,
+    fused: Tensor,
```

Comment:
```suggestion
    preds: Tensor,
    target: Tensor,
```
@ankitaS11 Please use `preds, target` naming for variables to stay consistent with the rest of `torchmetrics` package :]

Reply:
I've a question regarding this, @stancld - since this metric (`D_LAMBDA`) is a no-reference image metric, so we don't have a `target` per say... Do you still think that renaming it to `preds`/`target` is a good idea?

I believe it's a little confusing, but I also agree that we should stay consistent. So I don't have a strong opinion here, and if you say, I'm happy to rename it for now. Maybe, in the future, we can think of more generic names for all the metrics in the `torchmetrics` API. Just sharing my thoughts :)

cc: @SkafteNicki @justusschock